src/main/java/org/cloudburstmc/server/ServerProperties.java

```diff
@@ -0,0 +1,270 @@
+package org.cloudburstmc.server;
+
+import lombok.val;
+import org.cloudburstmc.server.level.Difficulty;
+import org.cloudburstmc.server.player.GameMode;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.util.Properties;
+
+/**
+ * represents the server.properties file on disk
+ */
+public class ServerProperties {
+
+    /**
+     * prepare the server.properties file during initialization
+     * it will create the file and save all the default values in if not exists
+     * @param path the path of the file
+     * @return prepared server.properties
+     */
+    public static ServerProperties prepareServerProperties(Path path) {
```

Comment:
This could just be named `load` or something.

Also, why in world would you want to use `val`!?

Reply:
oops，I saw lombok is enabled somewhere so I thought I can use it, sorry

the reason I deliberately choose prepare instead of load is because it would create the file if not exists, no one would anticipate a load to writing something right?

Reply:
You can use `val` i just dont see a point to it since its barely used anywhere else, should keep consistency.

Reply:
or perhaps we should just inline the prepare function back to the Server class because it is just a small snippet of init logic?